SampleViewer/SyntaxHighlighter/QMLHighlighter.cpp

```diff
@@ -91,49 +91,52 @@ QMLHighlighter::QMLHighlighter(QTextDocument* parent) :
 
     foreach (const QString& pattern, keywordPatternsOther)
     {
-        rule.pattern = QRegExp(pattern);
+        rule.pattern = QRegularExpression(pattern);
         rule.format = m_keywordOtherFormat;
         m_highlightingRules.append(rule);
     }
 
     m_singleLineCommentFormat.setForeground(Qt::darkGreen);
-    rule.pattern = QRegExp("^(?!\"([^\"']*)\")\\s*//[^\n]*");
+    rule.pattern = QRegularExpression("^(?!\"([^\"']*)\")\\s*//[^\n]*");
     rule.format = m_singleLineCommentFormat;
     m_highlightingRules.append(rule);
 
     m_multiLineCommentFormat.setForeground(Qt::darkGreen);
 
     m_quotationFormat.setForeground(Qt::darkGreen);
-    rule.pattern = QRegExp("(\"[^\"]+\"|'[^\']+')");
+    rule.pattern = QRegularExpression("(\"[^\"]+\"|'[^\']+')");
     rule.format = m_quotationFormat;
     m_highlightingRules.append(rule);
 
-    m_commentStartExpression = QRegExp("/\\*");
-    m_commentEndExpression = QRegExp("\\*/");
+    m_commentStartExpression = QRegularExpression("/\\*");
+    m_commentEndExpression = QRegularExpression("\\*/");
 }
 
 void QMLHighlighter::highlightBlock(const QString& text)
 {
     foreach (const HighlightingRule& rule, m_highlightingRules)
     {
-        QRegExp expression(rule.pattern);
-        int index = expression.indexIn(text);
-        while (index >= 0)
+        QRegularExpression expression(rule.pattern);
+        QRegularExpressionMatchIterator matchIterator = expression.globalMatch(text);
+        while (matchIterator.hasNext())
         {
-            int length = expression.matchedLength();
+            QRegularExpressionMatch match = matchIterator.next();
+            int index = match.capturedStart();
+            int length = match.capturedLength();
             setFormat(index, length, rule.format);
-            index = expression.indexIn(text, index + length);
         }
     }
     setCurrentBlockState(0);
 
     int startIndex = 0;
     if (previousBlockState() != 1)
-        startIndex = m_commentStartExpression.indexIn(text);
+        startIndex = m_commentStartExpression.match(text).capturedStart();
 
     while (startIndex >= 0)
     {
-        int endIndex = m_commentEndExpression.indexIn(text, startIndex);
+        QRegularExpressionMatch match = m_commentEndExpression.match(text, startIndex);
+
+        int endIndex = match.capturedStart();
```

Comment:
Just one comment.(Optional) This is the only instance where we break it up into a `QRegularExpressionMatch` then get the `index` afterwards. It doesn't look like we reuse `match` afterwards besides getting the index, so maybe just simplify it to one line like the rest?